### PR TITLE
[SC-375] Add arbitrum

### PR DESCRIPTION
**Pull request by hexonaut**
Dependant on #6 . Also requires modifications to the xchain-helpers repo: https://github.com/marsfoundation/xchain-helpers/pull/13 . Adds Arbitrum dsr oracle.

### CODE REVIEW

File: test/DSROracleIntegrationArbitrum.t.sol [hunk 1]
```diff
@@ -0,0 +1,37 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+pragma solidity ^0.8.0;
+
+import "./DSROracleXChainIntegrationBase.t.sol";
+
+import { ArbitrumDomain } from "xchain-helpers/testing/ArbitrumDomain.sol";
+
+import { DSROracleForwarderArbitrumOne } from "../src/forwarders/DSROracleForwarderArbitrumOne.sol";
+import { DSROracleReceiverArbitrum } from "../src/receivers/DSROracleReceiverArbitrum.sol";
```
Comment (lucas-manuel) [inline test/DSROracleIntegrationArbitrum.t.sol]: Align imports

File: test/DSROracleIntegrationArbitrum.t.sol [hunk 2]
```diff
@@ -0,0 +1,37 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+pragma solidity ^0.8.0;
+
+import "./DSROracleXChainIntegrationBase.t.sol";
```
Comment (lucas-manuel) [inline test/DSROracleIntegrationArbitrum.t.sol]: Order imports in terms of relative paths, so this goes at bottom

File: src/forwarders/DSROracleForwarderArbitrumOne.sol [hunk 3]
```diff
@@ -0,0 +1,27 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+pragma solidity ^0.8.0;
+
+import { XChainForwarders } from 'xchain-helpers/XChainForwarders.sol';
+import { DSROracleForwarderBase } from './DSROracleForwarderBase.sol';
```
Comment (lucas-manuel) [inline src/forwarders/DSROracleForwarderArbitrumOne.sol]: Separate

File: test/DSROracleIntegrationArbitrum.t.sol [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+import { ArbitrumDomain } from "xchain-helpers/testing/ArbitrumDomain.sol";
+
+import { DSROracleForwarderArbitrumOne } from "../src/forwarders/DSROracleForwarderArbitrumOne.sol";
+import { DSROracleReceiverArbitrum } from "../src/receivers/DSROracleReceiverArbitrum.sol";
+
+contract DSROracleIntegrationArbitrumTest is DSROracleXChainIntegrationBaseTest {
+
+    DSROracleForwarderArbitrumOne forwarder;
+    DSROracleReceiverArbitrum receiver;
```
Comment (lucas-manuel) [inline test/DSROracleIntegrationArbitrum.t.sol]: Align

File: test/DSROracleIntegrationArbitrum.t.sol [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+        mainnet.selectFork();
+
+        address expectedReceiver = computeCreateAddress(address(this), 5);
+        forwarder = new DSROracleForwarderArbitrumOne(address(pot), expectedReceiver);
+
+        remote.selectFork();
+
+        oracle = new DSRAuthOracle();
+        receiver = new DSROracleReceiverArbitrum(address(forwarder), oracle);
+        oracle.grantRole(oracle.DATA_PROVIDER_ROLE(), address(receiver));
```
Comment (lucas-manuel) [inline test/DSROracleIntegrationArbitrum.t.sol]:
```suggestion
        oracle   = new DSRAuthOracle();
        receiver = new DSROracleReceiverArbitrum(address(forwarder), oracle);
        
        oracle.grantRole(oracle.DATA_PROVIDER_ROLE(), address(receiver));
```

**Review by lucas-manuel — CHANGES_REQUESTED** [review]

File: test/DSROracleIntegrationArbitrum.t.sol (hunk 1 above)
Comment (hexonaut) [reply test/DSROracleIntegrationArbitrum.t.sol]: done

File: test/DSROracleIntegrationArbitrum.t.sol (hunk 2 above)
Comment (hexonaut) [reply test/DSROracleIntegrationArbitrum.t.sol]: done

File: src/forwarders/DSROracleForwarderArbitrumOne.sol (hunk 3 above)
Comment (hexonaut) [reply src/forwarders/DSROracleForwarderArbitrumOne.sol]: fixed

File: test/DSROracleIntegrationArbitrum.t.sol (hunk 4 above)
Comment (hexonaut) [reply test/DSROracleIntegrationArbitrum.t.sol]: fixed

File: test/DSROracleIntegrationArbitrum.t.sol (hunk 5 above)
Comment (hexonaut) [reply test/DSROracleIntegrationArbitrum.t.sol]: fixed

**Review by lucas-manuel — APPROVED** [review]

**Review by barrutko — APPROVED** [review]

**Pull request merged**